src/gui/guiPasswordChange.cpp

```diff
@@ -259,3 +265,39 @@ bool GUIPasswordChange::OnEvent(const SEvent &event)
 
 	return Parent ? Parent->OnEvent(event) : false;
 }
+
+std::string GUIPasswordChange::getNameByID(s32 id)
+{
+	switch (id) {
+	case ID_oldPassword:
+		return "old_password";
+	case ID_newPassword1:
+		return "new_password_1";
+	case ID_newPassword2:
+		return "new_password_2";
+	}
+	return "";
+}
+
+#ifdef __ANDROID__
+bool GUIPasswordChange::getAndroidUIInput()
+{
+	if (!hasAndroidUIInput())
+		return false;
+
+	gui::IGUIElement *e;
+	if (m_jni_field_name == "old_password")
+		e = getElementFromId(ID_oldPassword);
+	if (m_jni_field_name == "new_password_1")
+		e = getElementFromId(ID_newPassword1);
+	if (m_jni_field_name == "new_password_2")
+		e = getElementFromId(ID_newPassword2);
+	if (e)
+	{
+		std::string text = porting::getInputDialogValue();
+		e->setText(utf8_to_wide(text).c_str());
+	}
+	m_jni_field_name.clear();
```

Comment:
Why must this be cleared?
Probably also `return true;` in the `if` above?

Reply:
I am not sure about that, the code is based on the existing `guiFormSpecMenu` implementation https://github.com/minetest/minetest/blob/9855651c06f84f29780cf6b25a392ac841b4630c/src/gui/guiFormSpecMenu.cpp#L2252. AFAIK the return value of this is never really checked anyway, I believe the output is cleared since we currently need to poll this in the absence of proper callbacks. That should probably be fixed at some point but is well beyond the scope of this PR.